frontend/src/utilities/useEnableApplication.tsx

```diff
@@ -59,66 +65,115 @@ export const useEnableApplication = (
     let watchHandle: ReturnType<typeof setTimeout>;
     if (enableStatus.status === EnableApplicationStatus.INPROGRESS) {
       const watchStatus = () => {
-        getValidationStatus(appId)
-          .then((response) => {
-            if (!response.complete) {
-              watchHandle = setTimeout(watchStatus, 10 * 1000);
-              return;
-            }
-            setEnableStatus({
-              status: response.valid
-                ? EnableApplicationStatus.SUCCESS
-                : EnableApplicationStatus.FAILED,
-              error: response.valid ? '' : response.error,
+        if (isInternalRouteIntegrationsApp(internalRoute)) {
+          getIntegrationAppEnablementStatus(internalRoute)
+            .then((response) => {
+              if (!response.isInstalled && response.canInstall) {
+                watchHandle = setTimeout(watchStatus, 10 * 1000);
+                return;
+              }
+              if (response.isInstalled && !response.canInstall) {
```

Comment:
A nit here -- I saw this and didn't change it because we were looking at other reasons for things...

```suggestion
              if (response.isInstalled) {
```
It doesn't matter if you can or not install if you're already installed. It's likely something we can trim.

Reply:
Updated

Reply:
This was not updated -- it's still here.